server/calendar/calendar-controller.js

```diff
@@ -0,0 +1,66 @@
+var passport = require('./gCalConfig');
+var gcal = require('google-calendar');
+
+module.exports.signin = passport.authenticate('google', { scope: ['https://www.googleapis.com/auth/plus.login','https://www.googleapis.com/auth/calendar','https://www.googleapis.com/auth/calendar.readonly'] });
+
+module.exports.callback = passport.authenticate('google', { successRedirect: '/',failureRedirect: '/calendar/failure' });
+
+module.exports.logout = function (req, res) {
+  req.logout();
+  req.session.destroy();
+  res.redirect('/');
+};
+
+module.exports.isAuthenticated = function (req, res) {
+  res.send(req.isAuthenticated() ? true : false); 
+};
+
+module.exports.AddCalendarEvent = function (callback) {
+  var accessToken = user.accessToken;
+  var calendarId = user.calendarID;
+  gcal(accessToken).events.quickAdd(calendarId, text, function(err, data) {
+    if (err) {
+      callback({error:err});
+    }
+    callback({data: data});
```

Comment:
here (and on line 36) i think that the callback will get called on the data regardless of whether there is an error. is that the intended behavior?